### PR TITLE
docs: add retriever guide, address minor onboarding feedbacks & enhancement

### PR DESCRIPTION
**Pull request by AyushExel**
- Tried to address some onboarding feedbacks listed in https://github.com/lancedb/lancedb/issues/1224
- Improve visibility of pydantic integration and embedding API. (Based on onboarding feedback - Many ways of ingesting data, defining schema but not sure what to use in a specific use-case)
- Add a guide that takes users through testing and improving retriever performance using built-in utilities like hybrid-search and reranking
- Add some benchmarks for the above
- Add missing cohere docs

### CODE REVIEW

File: docs/src/guides/tuning_retrievers/2_reranking.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+| Query Type | Hit-rate@5 |
+| --- | --- |
+| Vector |  0.640 |
+| FTS   |  0.595  |
+| Reranked vector | 0.677    |
+| Reranked fts  | 0.672    |
+| Hybrid | 0.759 |
+
+### SQuAD Dataset
```
Comment (AyushExel) [inline docs/src/guides/tuning_retrievers/2_reranking.md:63]: This is left out on purpose. Need to rerun the numbers

**Comment by AyushExel** [comment]
lancedb jni test issue seems unrelated to this PR

**Review by westonpace — APPROVED** [review]
Very cool, I think this is a great description

File: docs/src/basic.md [hunk 2]
```diff
@@ -180,6 +180,9 @@ table.
 
 !!! info "Under the hood, LanceDB reads in the Apache Arrow data and persists it to disk using the [Lance format](https://www.github.com/lancedb/lance)."
 
+!!! info "Automatic vectorization with Embedding API"
+    When working with embedding models, it is recommended to use LanceDB embedding API to automatically vectorize the data and queries in the background. See the [quickstart example](#using-embedding-api) or the embedding API [guide](./embeddings/)
```
[the remark this answers is not in the document]
Comment (AyushExel) [reply docs/src/basic.md]: yeah its pretty common.. IG usage difference is "vectorizing a op/algo" vs "vectorizing data" .. Example https://neptune.ai/blog/vectorization-techniques-in-nlp-guide

[the remark this answers is not in the document]
Comment (AyushExel) [reply docs/src/basic.md]: Okay changed it to "create vector representation of the data"

File: docs/src/basic.md [hunk 3]
```diff
@@ -424,6 +430,21 @@ Use the `drop_table()` method on the database to remove a table.
...
+
+    ```python
+    --8<-- "python/python/tests/docs/test_embeddings_slow.py:imports"
+    --8<-- "python/python/tests/docs/test_embeddings_slow.py:openai_embeddings"
+    ```
+
+Learn about using the existing integrations and creating custom embedding functions in the [embedding API guide](./embeddings/).
+
+!!! note "Using Embedding API in Rust & Typescript"
+    The Embedding API is currently only available in Python. Support for Rust and Typescript is being worked on.
```
[the remark this answers is not in the document]
Comment (AyushExel) [reply docs/src/basic.md:446]: Okay I'll just remove this note for now and add rust and js examples when I come back

**Pull request merged**